Revising 5e content

lastis opened the pull request:
I did some changes to the 5e content I think other people also would benefit from. 

- Removed background flaws, ideals and bonds: Background flaws, ideals and bonds are typically tailored to the specific character. I saw no need to have these clutter up the tabs and the character sheet. 
-  Corrected the Sentinel feat errata.
-  Added a prefix to class feature names (Warlock and Paladin only so far) ex. "Paladin ~ Divine Smite" instead of "Divine Smite".
-  Changed Human race mod to appear in the race selection tab.
-  Changed Variant Human stat changes to be racial modifiers.
-  Replaced 99 with 0 in the number of spells per day for the lvl 0 spells.
-  Removed unnecessary tab spacings.
-  Added cantrips as 0 lvl spells. (They were commented out.)
-  Removed the dex modifier from AC when using heavy armor.

BahamutDragon commented:
Mate,

Where to begin... 
While I welcome PRs and changes, broad changes to standards really should merit a conversation. Some of the implementation exist for a specific reason. Such as no 0th level Spells. Adding that back in alters the dynamics, and does not follow the established publisher rules. I welcome your input on those features, but since I built the entire set from scratch, and my personal group uses these rules, I had the benefit of altering things to make it work for all involved.

Why would you want to remove the background option to choose Flaws, ideals, and bonds? Backgrounds are a package deal. If a person chooses to not use them there is no big deal, it will not appear on the sheet. If you remove it, then you have decided for everyone that they cannot use these options. I find it best to offer everything available, and let the person using the data to choose what they do with it. Taking away choices is not something I will agree with. If you want to discuss placing those options in a different location, then let us have that discussion.

Needless changes, such as removing tab spacing only affects the programmers, not the end users. This actually places more work on me if I need to fix an issue. Differentials can ignore whitespace changes, but when adding/altering things, the lined up nicely makes for faster catching of mistakes and keeps things organized. Tab space is set to 6, if you set your editor to that setting, you will see that those line up nicely in most of the files.

Some of these changes I welcome, they address specific problems. Lumping it all into one PR means I either accept or reject everything.

BahamutDragon commented on data/5e/wizards_of_the_coast/core/players_handbook/ph_classes.lst:
```diff
@@ -1,569 +1,569 @@
+CLASS:Barbarian	HD:12	TYPE:PC.BASE	MAXLEVEL:20	DEFINE:BarbarianLVL|0	BONUS:VAR|BarbarianLVL|CL				
+# Class Name	Multiple Requirements								
+CLASS:Barbarian	PREMULT:1,[!PRETEMPLATE:1,First Level],[PREABILITY:1,CATEGORY=Internal,Multiclass Allowed ~ Barbarian],[PRECLASS:1,Barbarian=1]								
+###Block									
```
Why are we adding so much whitespace?

BahamutDragon commented on data/5e/wizards_of_the_coast/core/players_handbook/ph_classes.lst:
```diff
@@ -1,569 +1,569 @@
+18	ABILITY:Monk Class Feature|AUTOMATIC|Monk ~ Empty Body								
+20	ABILITY:Monk Class Feature|AUTOMATIC|Monk ~ Perfect Self								
+									
+# Class Name	Hit Dice	Type	Define	Modify VAR	FACT				
```
If header comment lines do not line up why keep them?

BahamutDragon commented on data/5e/wizards_of_the_coast/core/players_handbook/ph_races.lst:
```diff
@@ -1,49 +1,49 @@
+																					
+# Basic Rules Races																					
+# Race Name	SORTKEY	Size	Move	Stat bonus	Ability	Nb Legs	Nb Hands	Main Race Type	Race Subtype	Type	Source Page										
+Dwarf	SORTKEY:Base_PC	SIZE:M	MOVE:Walk,25	BONUS:STAT|CON|2	ABILITY:Internal|AUTOMATIC|Racial Traits ~ Dwarf	LEGS:2	HANDS:2	RACETYPE:Humanoid	RACESUBTYPE:Dwarf	TYPE:Humanoid.PC	SOURCEPAGE:p.18										
```
Why are you making the upcoming conversion process harder? If you are volunteering to handle the entire 5e conversion to the new pcgen formula system, then feel free to remove these intermediate steps to making that easier for me. Arbitrarily removing code without understanding the reasoning is not a good practice though.

BahamutDragon commented on data/5e/wizards_of_the_coast/core/universal_rules/ur_abilities.lst:
```diff
@@ -1,281 +1,281 @@
-#	BONUS:VAR|ReplacesBackgroundFeature|1|TYPE=Boolean|PREFACT:1,ABILITIES,ReplacesBackgroundFeature=true
\ No newline at end of file
+# CVS $Revision: $ $Author: $ -- Mon Feb 22 22:56:28 2016 -- reformated by PCGen PrettyLST v6.06.00																			
+																			
```
I really do not understand the reasoning here... condense all the tab spaces where we have content, but then add filler white space behind everything else, including empty lines.

BahamutDragon commented on data/5e/wizards_of_the_coast/core/universal_rules/ur_abilities.lst:
```diff
@@ -1,281 +1,281 @@
+																			
+###Block																			
+# 5e does not use the exact same system as 3.5e.																			
+"# Different effects ""SET"" the base value and then bonuses alter that."																			
```
Why are we uncommenting a line here?

BahamutDragon commented on system/gameModes/5e/load.lst:
```diff
@@ -1,45 +1,38 @@
-SIZEMULT:T|4
```
The 5e gamemode file is obsolete as the pcgen release includes a fixed version. This should just be removed entirely to avoid conflicts.

BahamutDragon commented on data/5e/wizards_of_the_coast/supplement/elemental_evil_players_companion/eepc_races.lst:
```diff
@@ -1,5 +1,5 @@
-Aarakocra	SOURCEPAGE:p.3	SIZE:M	MOVE:Walk,0	ABILITY:Internal|AUTOMATIC|Racial Traits ~ Aarakocra	LEGS:2	HANDS:2	RACETYPE:Humanoid	RACESUBTYPE:Aarakocra	TYPE:Humanoid.PC	SORTKEY:Base_PC
-Genasi	SOURCEPAGE:p.7	SIZE:M	MOVE:Walk,0	ABILITY:Internal|AUTOMATIC|Racial Traits ~ Genasi	LEGS:2	HANDS:2	RACETYPE:Humanoid	RACESUBTYPE:Genasi	TYPE:Humanoid.PC	SORTKEY:Base_PC
-Goliath	SOURCEPAGE:p.10	SIZE:M	MOVE:Walk,0	ABILITY:Internal|AUTOMATIC|Racial Traits ~ Goliath	LEGS:2	HANDS:2	RACETYPE:Humanoid	RACESUBTYPE:Goliath	TYPE:Humanoid.PC	SORTKEY:Base_PC
+												
```
Needless churn from added whitespace

BahamutDragon commented on data/5e/wizards_of_the_coast/supplement/unearthed_arcana/waterborne_adventures/ua_races.lst:
```diff
@@ -1,3 +1,3 @@
-
-
-Minotaur (Krynn)		SOURCEPAGE:p.2		SIZE:M		MOVE:Walk,30	ABILITY:Internal|AUTOMATIC|Racial Traits ~ Minotaur (Krynn)	LEGS:2	HANDS:2	RACETYPE:Humanoid	RACESUBTYPE:Minotaur	TYPE:Humanoid.PC	SOURCEPAGE:p.2	SORTKEY:Base_PC	BONUS:STAT|STR|2
+													
```
Needless churn from added whitespace

lastis commented:
Hey!

Thanks for your reply. I did not truly expect the pr. to be accepted as I agree many of the changes are my personal preferences. I am more than happy to create separate pull requests for the less controversial changes while leaving changes such as lvl 0 spells to a different discussion. 

The follow points I think are fixes I think should be implemented:
- Corrected the Sentinel feat errata.
- Changed Human race mod to appear in the race selection tab.
- Changed Variant Human stat changes to be racial modifiers.
- Removed the dex modifier from AC when using heavy armor.

I have some opinions about the other changes, but I see no need to push them if the community does not support them. 

**Title Prefixes:** Adding prefixes to the title of special abilities I think is a good way to see where the abilities come from. Especially from racial features such as the extra skills from human. Without prefix the racial feature will appear as "Skills" vs. "Human ~ Skills". I realize this requires many entries to be changed for what is maybe only a minor issue. I'm happy to hear your input. 

**Flaws and bonds:** I agree removing flaws and bonds completely might be a step in the wrong direction. Keeping the choice available is a good argument. I simply removed them because it was the most simple solution I found to avoid them cluttering up the tabs and character sheet. This "issue" can probably be solved by organizing flaws and bonds differently. I

**Cantrips as lvl 0 spells:** Could you elaborate your position on this again? My reason for adding the cantrips as lvl 0 spells were simply because it did not make sense to me to distinguish them from other spells. When I was trying to implement some 3rd party cantrips it was easier to add them as lvl 0 spells than special abilities. It seem overly complicated when there was a simple solution. 

About the spacing, when I dived into this I had no idea how my editor could properly view the .lst files. Seeing they were tab separated values I simply used office libre calc to view them as spreadsheets. That is the reason behind the "formatting". How do you view these files? To be quite honest I find tab separated values to be quite an unwieldy format.

BahamutDragon commented:
These issues:

    Corrected the Sentinel feat errata.
    Changed Human race mod to appear in the race selection tab.
    Changed Variant Human stat changes to be racial modifiers.
    Removed the dex modifier from AC when using heavy armor.

Sound reasonable to me. Please separate them out for a proper review.

I use a text editor to edit the files. Since I pester the pcgen team, I tend to use what they use, which would be editplus. Easier to keep files in sync when they do all the work.

Cantrips as lvl 0 spells: One word "Consistency". In order for cantrip allotment to be gained in a consistent manner between Learned/Spellbook vs. Known casters, the granting method using abilities was chosen. Wizards can otherwise learn all the cantrips available. This was intentionally done so the user would know how many cantrips they get.

BahamutDragon commented:
I am a little curious why you think this:
   
    Title Prefixes: Adding prefixes to the title of special abilities I think is a good way to see where the abilities come from. Especially from racial features such as the extra skills from human. Without prefix the racial feature will appear as "Skills" vs. "Human ~ Skills". I realize this requires many entries to be changed for what is maybe only a minor issue. I'm happy to hear your input. 

would be needed? The new PCGen Output Sheet separates out Race granted features and Class Features. Furthermore, I sorted the class features based upon their base class. I do not anticipate many multiclass players, but that should suffice.

We should take these design discussions to either the BD Games yahoo group for general discussion by those that participate, or email me directly if you wish.

lastis commented:
After seeing the interaction between cantrips and wizard I understand why the current method was chosen.  Title prefixes I think would simply look nicer, but I wont bother changing every entry anyway so I will simply leave it as it is now.
